foundation/org.eclipse.persistence.core/src/org/eclipse/persistence/queries/ReportQueryResult.java

```diff
@@ -151,6 +124,44 @@ protected void buildResult(ReportQuery query, AbstractRecord row, Vector toManyD
         setResults(results);
     }
 
+    private Object processConstructorItem(ReportQuery query, AbstractRecord row, Vector toManyData, ConstructorReportItem constructorItem) {
+        // For constructor items need to process each constructor argument.
+        Class[] constructorArgTypes = constructorItem.getConstructorArgTypes();
+        int numberOfArguments = constructorItem.getReportItems().size();
+        Object[] constructorArgs = new Object[numberOfArguments];
+
+        for (int argumentIndex = 0; argumentIndex < numberOfArguments; argumentIndex++) {
+            ReportItem argumentItem = (ReportItem)constructorItem.getReportItems().get(argumentIndex);
+            Object result = null;
+            if(argumentItem.isConstructorItem()) {
+                result = processConstructorItem(query, row, toManyData, (ConstructorReportItem) argumentItem);
+            } else {
+                result = processItem(query, row, toManyData, argumentItem);
+            }
+            constructorArgs[argumentIndex] = ConversionManager.getDefaultManager().convertObject(result, constructorArgTypes[argumentIndex]);
+        }
+        try {
+            Constructor constructor = constructorItem.getConstructor();
+            Object returnValue = null;
+            if (PrivilegedAccessHelper.shouldUsePrivilegedAccess()){
+                try {
+                    returnValue = AccessController.doPrivileged(new PrivilegedInvokeConstructor(constructor, constructorArgs));
+                } catch (PrivilegedActionException exception) {
+                    throw QueryException.exceptionWhileUsingConstructorExpression(exception.getException(), query);
+                }
+            } else {
+                returnValue = PrivilegedAccessHelper.invokeConstructor(constructor, constructorArgs);
+            }
+            return returnValue;
+        } catch (IllegalAccessException exc){
+            throw QueryException.exceptionWhileUsingConstructorExpression(exc, query);
+        } catch (java.lang.reflect.InvocationTargetException exc){
+            throw QueryException.exceptionWhileUsingConstructorExpression(exc, query);
+        } catch (InstantiationException exc){
```

Comment:
what about catching ReflectiveOperationException instead of 3 of its specific subtypes?

Reply:
...ah, just moving the code around... worth to be updated but its nothing 'must have'...

Reply:
Sure, I can change this while I'm in there.